File: sdks/python/src/boxsdk/models/shared/group.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils

class GroupGroupTypeEnum(str, Enum):
    MANAGED_GROUP = "managed_group"
    ALL_USERS_GROUP = "all_users_group"

class GroupTypeEnum(str, Enum):
    GROUP = "group"


@dataclass_json
@dataclasses.dataclass
class Group:
    r"""Group
    A base representation of a group.
    """
    
    created_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('created_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    group_type: Optional[GroupGroupTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('group_type') }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    modified_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('modified_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    type: Optional[GroupTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    

File: sdks/ts/src/sdk/models/shared/group.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum GroupGroupTypeEnum {
    ManagedGroup = "managed_group",
    AllUsersGroup = "all_users_group"
}

export enum GroupTypeEnum {
    Group = "group"
}


// Group
/** 
 * A base representation of a group.
**/
export class Group extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=created_at" })
  createdAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=group_type" })
  groupType?: GroupGroupTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=modified_at" })
  modifiedAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: GroupTypeEnum;
}
